community/server/src/main/java/org/neo4j/server/web/HttpHeaderUtils.java

```diff
@@ -49,4 +54,30 @@ public static MediaType mediaTypeWithCharsetUtf8(MediaType mediaType)
         paramsWithCharset.putAll(CHARSET);
         return new MediaType(mediaType.getType(), mediaType.getSubtype(), paramsWithCharset);
     }
+
+    /**
+     * Retrieve custom transaction timeout in milliseconds from numeric {@link #MAX_EXECUTION_TIME_HEADER} request
+     * header.
+     * If header is not set returns -1.
+     * @param request http request
+     * @param errorLog errors log for header parsing errors
+     * @return custom timeout if header set, -1 otherwise or when value is not a valid number.
+     */
+    public static long getTransactionTimeout( HttpServletRequest request, Log errorLog )
+    {
+        String headerValue = request.getHeader( MAX_EXECUTION_TIME_HEADER );
+        if ( headerValue != null )
+        {
+            try
+            {
+                return Long.parseLong( headerValue );
+            }
+            catch ( NumberFormatException e )
+            {
+                errorLog.error( String.format( "Fail to parse `%s` header with value: '%s'. Should be a positive number.",
+                        MAX_EXECUTION_TIME_HEADER, headerValue), e );
+            }
+        }
+        return -1;
```

Comment:
Ping about this one. By the looks of it, the usages of this value never checks for -1, only `> 0` so I think it makes sense to use 0, potentially also with something like a `UNSPECIFIED_TIMEOUT` constant.